### PR TITLE
Connect app to Saleor

### PR DESCRIPTION
**Pull request by orzechdev**
It adds in dashboard app:
- connection with Saleor
- persisting settings in app metadata
- update types
- fix typing errors
- fix form handling

### CODE REVIEW

File: apps/checkout-app/constants.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const APP_DOMAIN = process.env.NEXT_PUBLIC_VERCEL_URL;
+export const APP_URL = `http://${APP_DOMAIN}`;
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply apps/checkout-app/constants.ts:2]: But it doesn't include http or https protocol prefix

File: apps/checkout-app/frontend/components/templates/ChannelDetails/ChannelDetails.tsx [hunk 2]
```diff
@@ -15,26 +16,59 @@ import {
...
+import { Item } from "types/common";
+import Skeleton from "@material-ui/lab/Skeleton";
+import AppSavebar from "@frontend/components/elements/AppSavebar";
+import { Controller, useForm } from "react-hook-form";
+import { getActivePaymentProvider, getFormDefaultValues } from "./data";
+import { useEffect } from "react";
+import { ChannelsQuery } from "@graphql";
 
 interface ChannelDetailsProps {
   channelPaymentOptions?: ChannelPaymentOptions;
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply apps/checkout-app/frontend/components/templates/ChannelDetails/ChannelDetails.tsx]: I have changed it to required, but I also added fallback return to `ErrorDetails` in page component if there is no `channelPaymentOptions`. If we want to have error boundary, the error has to be thrown manually, but then `channelPaymentOptions` should be optional and thrown in the details component if it is undefined.

File: apps/checkout-app/constants.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jwm0) [reply apps/checkout-app/constants.ts:2]: This is fine, I'm just talking about referencing VERCEL_URL when assigning NEXT_PUBLIC_VERCEL_URL, we could make an `.env.production` and just put it there as well

[the remark this answers is not in the document]
Comment (orzechdev) [reply apps/checkout-app/constants.ts:2]: ohh, ok, sure I'll add it

File: apps/checkout-app/frontend/components/templates/ChannelDetails/ChannelDetails.tsx [hunk 3]
```diff
@@ -45,71 +80,114 @@ const ChannelDetails: React.FC<ChannelDetailsProps> = ({
...
+                                <Switch
+                                  name={flattenSettingId(
+                                    paymentOptionIdx,
+                                    provider.id
+                                  )}
+                                  checked={field.value === provider.id}
+                                  onChange={() =>
+                                    field.onChange({
+                                      target: {
+                                        name: paymentOption.id,
```
Comment (bmigirl) [inline apps/checkout-app/frontend/components/templates/ChannelDetails/ChannelDetails.tsx:161]: This correct? Should name and id be the same thing?

File: apps/checkout-app/frontend/data.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  activePaymentProviders?: ChannelActivePaymentProviders
+): ChannelPaymentOptions[] =>
+  channels.map((channel) => ({
+    id: channel.id,
+    channel: channel,
+    paymentOptions: paymentMethods.map((method) => ({
+      id: method.id,
+      method,
+      availableProviders: paymentProviders,
+      activeProvider: activePaymentProviders?.[channel.id]?.[method.id]
```
Comment (bmigirl) [inline apps/checkout-app/frontend/data.ts:71]: Can we make some local consts in this function? I think activeProvider is a bit difficult to read, maybe put this part in a const 
```findById(
            paymentProviders,
            activePaymentProviders[channel.id][method.id]
          ) || null

**Review by bmigirl — CHANGES_REQUESTED** [review]

File: apps/checkout-app/frontend/components/templates/ChannelDetails/ChannelDetails.tsx (hunk 3 above)
Comment (orzechdev) [reply apps/checkout-app/frontend/components/templates/ChannelDetails/ChannelDetails.tsx:161]: Yes, ids differentiate the fields here

**Review by jwm0 — APPROVED** [review]

File: apps/checkout-app/frontend/data.ts (hunk 4 above)
Comment (orzechdev) [reply apps/checkout-app/frontend/data.ts]: done

**Pull request merged**